22 build summary detail amt

allison-truhlar opened the pull request:
### What changes did I make?
- Added the SummaryDetail component and associated child components (e.g., ingredients, instructions)
- Added BackgroundBlur component that appears behind SummaryDetail. Also, if you click on it, the SummaryDetail will close.
- Added basic Tailwind styling to the SummaryDetail component and associated child components

### Why did I make these changes?
- When the user clicks on "view recipe details," the project requires that a Summary Detail component with the detailed recipe information appear.

### Known issues
I'm opening this PR with a few known issues because I'm stuck on them, and they're more styling-related, so I'm going to open a new issue for styling this component and add these items in the issue:
- The close button in SummaryDetail refuses to stay in the upper right corner! I had it working and then something must have changed in the code when I was working on other thing because it no longer stays.
- SummaryDetail does not "slide" in and out, despite my best attempts with the CSS. It just appears/disappears.
- I can't get the picture at the top of SummaryDetail to spread across the entire width of the component and zoom in on the image, without increasing the height of the div it's contained in. If it fills the width, it often exceeds the height of the div and overflows into the next div.

### Screenshots

<details><summary>Top of the SummaryDetail component</summary>

![Screenshot 2023-10-27 at 2 26 50 PM](https://github.com/chingu-voyages/v46-tier2-team-20/assets/114596142/3df1c8b3-5919-4ea0-b5b5-40d5aacbec38)

</details> 

<details><summary>Bottom of the SummaryDetail component (a little bit of the middle is cutoff between the two images)</summary>

![Screenshot 2023-10-27 at 2 26 57 PM](https://github.com/chingu-voyages/v46-tier2-team-20/assets/114596142/efc164fc-e570-4224-9dcf-e8cff05570bc)

</details>

allison-truhlar committed:
Add button to RecipeBrief component

The button passes the id of the selected recipe to the function handleRecipeBriefClick.

allison-truhlar committed:
Renamed RecipeDisplay folder to recipeBrief

This is aligned with our naming convention (lowercase folders) and also convention of co-locating a single component and component style file in one folder

allison-truhlar committed:
Create SummaryDetail.jsx

Created an empty SummaryDetail component

allison-truhlar committed:
Update App.jsx

Import SummaryDetail component. Moved the handleRecipeBriefClick function up from the RecipeBrief component, and then passed it down as props to RecipeBrief

allison-truhlar committed:
Update RecipeBrief.jsx

Removed handleRecipeBrief click function - moved it up to App.jsx

allison-truhlar committed:
Changed handleRecipeBriefClick in RecipeBrief.jsx

The function now passes the whole recipe back to App.jsx, not just the id

allison-truhlar committed:
Added recipeDetail state to App.jsx

The recipe passed back by handleRecipeBriefClick to App.jsx now gets set as recipeDetail in App.jsx, and then passed down to the SummaryDetail component

allison-truhlar committed:
Set up SummaryDetail component to accept recipeDetail as props

allison-truhlar committed:
Added state isDetailShown

To toggle on/off whether the summaryDetail component is shown

allison-truhlar committed:
Removed id from the recipe deconstruction in RecipeBrief

No longer needed

allison-truhlar committed:
Added destructuring of recipeDetail data in SummaryDetail component

Created ingredientEls and nutritionEls

allison-truhlar committed:
Added instructionEls to SummaryDetail

allison-truhlar committed:
Added categoryEls

allison-truhlar committed:
Created new child components for SummaryDetail

Moved the Ingredients, Instructions, Nutrition, and Categories out of SummaryDetail into their own components, which are child components to SummaryDetail

allison-truhlar committed:
Added a toggleIsDetailShown function

Toggles whether the isDetailShown state is true or false. Hires when clicking on a recipe name or when clicking on the close button in SummaryDetail

allison-truhlar committed:
Added styling and handleSummaryDetailClose

The function handleSummaryDetailClose both toggles isDetailShown and sets recipeDetail back to null

allison-truhlar committed:
Adding basic Tailwind styling

allison-truhlar committed:
Added styling to bring in SummaryDetail from right

allison-truhlar committed:
Added sorting and capitalize function to Nutrition

Now the nutrition elements are all capitalized and dispalyed in alphabetcial order

allison-truhlar committed:
Added how to video link to SummaryDetail

allison-truhlar committed:
Added BackgroundBlur component

allison-truhlar committed:
Add toggleNutrition, and icons

allison-truhlar committed:
If you click on BackgroundBlur, SummaryDetail will close

komalgill0310 commented on src/components/summaryDetail/Ingredients.jsx:
```diff
@@ -0,0 +1,11 @@
+
+export default function Ingredients({ sections }) {
+  const ingredientEls = sections.map((section) => section.components.map((component) => (<p key={nanoid()} className="pb-2 text-xs">{component.raw_text}</p>)));
+
```
As Robester mentioned in the code review of my code, "I assume nanoid is a random key generator. It is not recommended to use a random key generator as a key. You should use some sort of combination of strings from the recipes object or a unique id provided by the backend, which should also be in the recipes object."

allison-truhlar replied:
Addressed in commit [9104615](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/9104615e94971216a6810332980e97b2eacc012a), below

komalgill0310 commented on src/components/backgroundBlur/BackgroundBlur.jsx:
```diff
@@ -0,0 +1,8 @@
+import "./BackgroundBlur.css"
+
+export default function backgroundBlur({handleSummaryDetailClose}) {
```
Function names should start with Capital Letters.

allison-truhlar replied:
Resolved with commit [997c989](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/997c98971aaf211d9e518c3dc124781e917bcad0), below

komalgill0310 commented on src/components/summaryDetail/SummaryDetail.jsx:
```diff
@@ -0,0 +1,77 @@
+import Categories from './Categories';
+
+export default function SummaryDetail({
+  recipeDetail, setRecipeDetail, isDetailShown, toggleIsDetailShown, handleSummaryDetailClose,
```
1. setRecipeDetail and toggleIsDetailShown are not being used in the logic, so I guess there is no need for them to pass as a function parameter.

allison-truhlar replied:
Addressed in commit [8eb5dbd](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/8eb5dbdf846190aaed62d1d8c0df0ec0033b39a5), below

komalgill0310 commented on src/components/summaryDetail/Nutrition.jsx:
```diff
@@ -0,0 +1,29 @@
+    .filter(([key, value]) => key !== 'updated_at')
+    .sort(([keyA], [keyB]) => keyA.localeCompare(keyB))
+    .map(([key, value]) => (
+      <p key={nanoid()} className="border-t text-xs">
```
As Robester mentioned in the code review of my code, "I assume nanoid is a random key generator. It is not recommended to use a random key generator as a key. You should use some sort of combination of strings from the recipes object or a unique id provided by the backend, which should also be in the recipes object."

allison-truhlar replied:
Addressed with commit [6148bb1](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/6148bb1f4f35ee60f226aa0f13b608da3f5975ad), below

komalgill0310 commented on src/components/summaryDetail/Instructions.jsx:
```diff
@@ -0,0 +1,17 @@
+
+export default function Instructions({ instructions }) {
+  const instructionEls = instructions.map((instruction, index) => (
+    <p key={nanoid()} className="pt-4 text-xs">
```
As Robester mentioned in the code review of my code, "I assume nanoid is a random key generator. It is not recommended to use a random key generator as a key. You should use some sort of combination of strings from the recipes object or a unique id provided by the backend, which should also be in the recipes object."

allison-truhlar replied:
Addressed with commit [919f74e](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/919f74e0503b9e9cbe3044503f0b9f3e9f13b4f4)

komalgill0310 commented on src/components/summaryDetail/Categories.jsx:
```diff
@@ -0,0 +1,14 @@
+import { nanoid } from 'nanoid';
```
this import is unnecessary as the logic does not use nanoid.

allison-truhlar replied:
Addressed with commit [a118fed](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/a118fed58e19eaa879bff38aa690a26fb0bbcc39)

komalgill0310 reviewed (changes requested):
I will discuss the CSS changes with @AOA19 and the known issues.

allison-truhlar committed:
Changes to remove nanoid from Ingredients.jsx

I removed the nanoid package from Ingredients.jsx. To replace nanoid for the ingredient key, I passed the recipeId from SummaryDetail to Ingredients. The key for each ingredient is now a combination of the recipe id and the ingredient id.

allison-truhlar committed:
Removed import of nanoid package from Ingredients.jsx

allison-truhlar committed:
Changed backgroundBlur to BackgroundBlur

Functions should have a capital letter for the first letter

allison-truhlar committed:
Removed unneeded props from SummaryDetail

setRecipeDetail and toggleIsDetailShown were not used in SummaryDetail, so I removed them as props passed from App.jsx to SummaryDetail.jsx

allison-truhlar committed:
No longer use nanoid for key in Nutrition.jsx

REmoved nanoid package from Nutrition.jsx. Added recipeId as a prop passed from SummaryDetail to Nutrition. Now the key for each nutrition element in Nutrition is made up of recipeId + the name of the nutrition key (e.g., calories, carbs, fat, etc.)

allison-truhlar committed:
removed nanoid from Instructions.jsx

Removed nanoid package import from Instructions.jsx. Added recipeId as a prop passed from SummaryDetail to Instructions. Now the key for each instruction element is made up of the recipeId + the instruction id

allison-truhlar committed:
Removed nanoid import from Categories.jsx

This import was not used in the Categories.jsx logic

allison-truhlar commented:
Hi @komalgill0310! Thank you for your review! I made all the changes you suggested - I included a comment after each of your comments that indicates which commit you can find the change in. Please let me know if you have any more changes to suggest!

komalgill0310 committed:
Merge branch 'development' into 22-build-summary-detail-AMT
